### PR TITLE
[AIR] Update `TorchPredictor` to new Predictor API

### PR DESCRIPTION
**Pull request by amogkam**
Depends on #25534 

Updates and cleans up `TorchPredictor`
- uses `predict_pandas` API
- removes unnecessary args from `predict`.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/air/predictors/integrations/torch/torch_predictor.py [hunk 1]
```diff
@@ -45,81 +46,54 @@ def from_checkpoint(
...
-    def _predict(self, tensor: torch.Tensor) -> pd.DataFrame:
-        """Handle actual prediction."""
-        prediction = self.model(tensor).cpu().detach().numpy()
-        # If model has outputs a Numpy array (for example outputting logits),
-        # these cannot be used as values in a Pandas Dataframe.
-        # We have to convert the outermost dimension to a python list (but the values
-        # in the list can still be Numpy arrays).
-        return pd.DataFrame({"predictions": list(prediction)}, columns=["predictions"])
+
+        self.model.eval()
```
Comment (krfricke) [inline python/ray/air/predictors/integrations/torch/torch_predictor.py:70]: do we need `with torch.no_grad()` here?

**Review by krfricke — APPROVED** [review]
TorchPredictor LGTM

File: python/ray/air/predictors/integrations/torch/torch_predictor.py (hunk 1 above)
Comment (amogkam) [reply python/ray/air/predictors/integrations/torch/torch_predictor.py]: Added!

Comment (amogkam) [reply python/ray/air/predictors/integrations/torch/torch_predictor.py]: Seems like we want both torch.no_grad and model.eval https://stackoverflow.com/questions/55627780/evaluating-pytorch-models-with-torch-no-grad-vs-model-eval

**Comment by amogkam** [comment]
@krfricke pinging if you want to take another look

**Review by Yard1 — APPROVED** [review]

**Pull request merged**